content/types/_client_options.textile

```diff
@@ -40,6 +40,8 @@ h4.
 
 - <span lang="default">environment</span><span lang="csharp,go">Environment</span><span lang="ruby">:environment</span> := _<span lang="default">null</span><span lang="java">Null</span><span lang="python">None</span><span lang="ruby,objc,swift">nil</span>_ Allows a "custom environment":https://support.ably.io/a/solutions/articles/3000077619, region or cluster to be used with the Ably service. Please "contact us":https://www.ably.io/contact if you require a custom environment. Note that once a custom environment is specified, the "fallback host functionality":https://support.ably.io/solution/articles/3000044636-routing-around-network-and-dns-issues is disabled by default.<br>__Type: @String@__
 
+- <span lang="default">idempotentRestPublishing</span><span lang="csharp,go">IdempotentRestPublishing</span><span lang="ruby">:idempotent_rest_publishing</span> := _false_ When true, enables idempotency, which attaches a message ID from the client side, allowing the Ably servers to ignore duplicate messages.<br>__Type: @Boolean@__
```

Comment:
"When true, enables idempotent publishing by assigning a unique message ID client-side, allowing the Ably servers to discard automatic publish retries following a failure such as a network fault. We recommend you enable this by default.  In version 1.2 onwards, idempotent publishing for retries will be enabled by default."